include/h2o/memory.h

```diff
@@ -301,6 +305,13 @@ void h2o_append_to_null_terminated_list(void ***list, void *element);
 
 /* inline defs */
 
+inline void *h2o_memcpy(void *dst, const void *src, size_t n)
+{
+    if (src)
```

Comment:
and let's do 'if ... h2o_fatal' instead of assert so that we'd see a crash even if the binary was built without assertion.